### ISSUE TITLE
Use PostCSS parser?

### ISSUE THREAD

**Issue by ben-eb** [frontendfriends/node-combine-mq#11]
PostCSS is being used by quite a few projects, the main one being https://github.com/postcss/autoprefixer - its parser is much faster than others and is very robust; but I think the main benefit is for usage in a pipeline of in-memory transforms, so that people can run tasks such as autoprefixing, combining media queries, minifying etc, without having to read from disk in every step.

The responsibility of disk operations can then be extracted out of the main source code of the module, so that it can be used as a plugin to operate on in-memory CSS.

What do you think? :smile:

https://github.com/postcss/postcss

**Comment by furzeface** [comment]
Hey @ben-eb - yep that's a great idea, one of the guys had a go at moving it over a while ago but we ran out of time. 

Is it something you'd want to look at helping out with? If not, we'll try and get round to it when there's time.

Thanks,
Dan.

**Comment by ben-eb** [comment]
Yeah, I'd like to help out with this; but, I am currently working through a lot of university work and only really have been publishing modules that are relevant to my assignments. So I am hopefully looking at doing more contributions to other projects later on in May, after my deadlines.

I had a look at https://github.com/hail2u/node-css-mqpacker which has the same goals as your project, but I wasn't sure how much overlap/difference there is between your two projects. Perhaps it might be good to join forces with @hail2u?

**Comment by ben-eb** [comment]
@furzeface I propose the following roadmap:

* Extract disk operations logic out of the main core of the module, so that we have three parts - disk IO, CSS transform, and CLI.
* Combine CLI and disk IO together, preferably using some stdio transforms to make the CLI more robust.
* Test the CSS transform standalone, with no disk IO. I propose using tape (https://github.com/substack/tape), a lightweight test runner.
* Clean up redundant code in the module, keeping the tests passing.
* Convert over to PostCSS.
* Possibly in there somewhere we can have a look at ES6 with Babel. If you are OK with that.

We will then have a module which can be consumed inside PostCSS, or as a standalone lib with a stdio supported CLI, and sourcemaps. Once this is done, I can have a look at bundling it with https://github.com/ben-eb/cssnano - I think combining media queries is an important minification step.

**Comment by furzeface** [comment]
A good way forward, having a look at your fork now too :) 

cc @spacedawwwg and @JohnCashmore too for this - you guys know way more about the original, want to get involved when you can?

**Comment by ben-eb** [comment]
Any updates on this?

**Comment by ben-eb** [comment]
I'm still happy to work on this but I need feedback from you guys.

**Comment by JohnCashmore** [comment]
I'd do whatever you thinks best a lot has changed in the front end world in
the past 12 months and I think working out if this is still a required tool
and what is the best tool to process the css may have changed
